blueman/Sdp.py

```diff
@@ -365,17 +366,19 @@
 
 
 class ServiceUUID(UUID):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, uuid: str):
+        super().__init__(uuid)
 
     @property
-    def short_uuid(self):
+    def short_uuid(self) -> Optional[int]:
         if self.reserved:
             return self.int >> 96 & 0xFFFF
+        else:
+            return None
 
     @property
-    def name(self):
-        if self.reserved:
+    def name(self) -> str:
+        if self.short_uuid:
```

Comment:
reserved -> short_uuid? If this is a bug please do a separate commit if typo please fix :smile:

Reply:
This is a trick to avoid a type error two lines below.

If `reserved` is `True` then `short_uuid` is an `int`, if it's `False` then it's `None`. So both basically express the same thing (neglecting the "impossible" edge case `uuid == 0`).

However, for the type checker it makes a difference: If we use `reserved` here, `short_uuid` two line above is of type `Optional[int]`. If we use `short_uuid` here, we know that it's not `None`, so it is of type `int`.